doc/java_parser.md

```diff
@@ -0,0 +1,67 @@
+---
+title: JavaParser 
+keywords: JavaParser
+---
+
+Here is JavaParser github project: [JavaPaser](https://github.com/javaparser/javaparser)
+
+## Differences between JavaParser and Spoon
+
+Spoon and [JavaParser](https://github.com/javaparser/javaparser) can both be used for analyzing and transforming Java source code.
+
+###Parsing
+Spoon is using Eclipse JDK compiler to build AST of sources, then Spoon AST is then made from compiler AST.
+JavaParser parses java source on their own. So JavaParser is faster, but all the heavy work (resolving of fully qualified type names, detection of implicit elements, ...) is done by JavaParser (@ftomassetti could you add there some comment about status)
+
+###AST model
+JavaParser AST is near to source code representation. You can found AST nodes like `EnclosedExpr`, which represents brackets like `(x+y)`or `Name`, which represents any java identifier. 
+Spoon's AST is pure java semantic model. It means it contains nodes, which represent the same **concepts**, which are used by java programmer when s/he thinks about java program. Spoon AST contains also implicit elements, which are not visible in java source at all. For example this code
+
+```java
+if (something) 
+	list = (List<String>) new ArrayList<>(FIELD_COUNT);
+```
+
+and this code
+
+```java
+if (something) {
+	OuterType.this.list = (((java.util.List<java.lang.String>) new java.util.ArrayList<java.lang.String>(Constants.FIELD_COUNT));
+}
+```
+
+has exactly same AST node elements. The only difference is that some nodes of AST model of first source code are marked as `implicit` 
+
+
+Comparison about common features:
+
+| Aspect | Spoon | JavaParser |
+| ------ | ------- | ------------ |
+| Declaration | Native navigation from [usage to declaration](http://spoon.gforge.inria.fr/references.html) | Supported in the [symbol resolver](https://github.com/javaparser/javasymbolsolver) |
+|License| [CeCILL-C](http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html) | LGPL License or Apache License |
+|Comments| Stored and attached to elements | Stored and attached to elements |
+
+Unique features of JavaParser
+
+* JavaParser is much faster with building of AST. 
+* JavaParser is lighter. Spoon has transitive dependencies to Eclipse JDK compiler and others. 
+
+
+Unique features in Spoon:
+
+* Spoon provides a way to semantically analyze code elements (methods `isSubType`, `isOverriding`) including correct resolving of generic arguments on methods and types.

```

Comment:
We had commented on this and other features to explain that JP has those too, since it was integrated ith JSS

Reply:
OK, So I remove `isSubtype` from list of differences. What about method isOverriding and **generics**?
```java
class A<T> {
<U> T method(U param1, T param2);
}
class B extends A<String> {
<Z> String method(Z 1, String p2);
}
```
Is it supported too by JP? ... it was quite challenging task even with all the information we already have in Spoon model.